source/image/ImageBuffer.cpp

```diff
@@ -169,26 +182,34 @@ void ImageBuffer::ShrinkToHalfSize()
 
 
 
-bool ImageBuffer::Read(const ImageFileData &data, int frame)
+int ImageBuffer::Read(const ImageFileData &data, int frame)
 {
 	// First, make sure this is a supported file.
 	bool isPNG = PNG_EXTENSIONS.contains(data.extension);
 	bool isJPG = JPG_EXTENSIONS.contains(data.extension);
+	bool isAVIF = AVIF_EXTENSIONS.contains(data.extension);
 
-	if(!isPNG && !isJPG)
+	if(!isPNG && !isJPG && !isAVIF)
 		return false;
 
-	if(isPNG && !ReadPNG(data.path, *this, frame))
-		return false;
-	if(isJPG && !ReadJPG(data.path, *this, frame))
-		return false;
+	int startFrame = frame;
+	int endFrame = startFrame;
```

Comment:
Why are you adding startFrame here if you're subtracting it at the end anyway?

Reply:
It was done that way to make the refactor easier, but isn't necessary anymore. I replaced it with a regular counter.